app/views/courses/TeacherClassView.coffee

```diff
@@ -226,6 +231,18 @@ module.exports = class TeacherClassView extends RootView
     else if @state.get 'selectedCourseInstance'
       @state.set 'selectedCourseInstance', null
 
+  getSelectedCourseInstance: ->
+    unless @state.get 'selectedCourseInstance'
+      @setSelectedCourseInstance()
+    return @state.get 'selectedCourseInstance'
+
+  setSelectedCourseLockedLevel: (level) ->
+    return unless me.showCourseProgressControl()
+    courseInstance = @getSelectedCourseInstance()
+    if courseInstance and level
+      courseInstance.set 'startLockedLevel', level
```

Comment:
@wjllance @AndrewJakubowicz we should not be using level slugs as a constant identifier because they can change.  We should be using level original ids in this type of scenario instead.